### PR TITLE
Fix/refs to inner objects

### PR DESCRIPTION
**Pull request by odersky**
Several formerly disabled tests are brought back. One required fixes of references to inner objects. Review by @DarkDimius

### CODE REVIEW

File: src/dotty/tools/dotc/ast/tpd.scala [hunk 1]
```diff
@@ -326,23 +326,24 @@ object tpd extends Trees.Instance[Type] with TypedTreeInfo {
...
-          }
-        prefix.select(tp)
+      case pre: SingletonType => correctForPath(singleton(pre)).select(tp)
       case pre => SelectFromTypeTree(TypeTree(pre), tp)
     } // no checks necessary
 
   def ref(sym: Symbol)(implicit ctx: Context): Tree =
     ref(NamedType(sym.owner.thisType, sym.name, sym.denot))
 
+  private def correctForPath(t: Tree)(implicit ctx: Context) = t match {
```
Comment (DarkDimius) [inline src/dotty/tools/dotc/ast/tpd.scala]: What does `For` in the name stand for? 
As this method is used only in `ref` and is private anyway, maybe it's better to move its definition to `ref` and give it a proper name, eg `followOuterLinks`?

**Comment by DarkDimius** [comment]
Otherwise LGTM

**Comment by odersky** [comment]
Since I am in train bringing back disabled tests, here's one more. AFAIK that's all the tests we had to disable in the past weeks, so we should be good on that side for now.

**Comment by DarkDimius** [comment]
LGTM

**Pull request merged**